src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1139,14 +1139,45 @@ public static bool IsParameterTypeContext(this SyntaxTree syntaxTree, int positi
         {
             tokenOnLeftOfPosition = tokenOnLeftOfPosition.GetPreviousTokenIfTouchingWord(position);
 
-            if (tokenOnLeftOfPosition.IsKind(SyntaxKind.OpenParenToken))
+            if (tokenOnLeftOfPosition.Parent.IsKind(SyntaxKind.ParenthesizedExpression,
+                    SyntaxKind.TupleExpression, SyntaxKind.TupleType) ||
+                tokenOnLeftOfPosition.Parent.IsParentKind(SyntaxKind.ParenthesizedExpression,
+                    SyntaxKind.TupleExpression, SyntaxKind.TupleType))
             {
-                return tokenOnLeftOfPosition.Parent.IsKind(SyntaxKind.ParenthesizedExpression,
-                    SyntaxKind.TupleExpression, SyntaxKind.TupleType);
+                return true;
+            }
+
+            // (int a, i$$
+            // ($$int a, int b
+            if (tokenOnLeftOfPosition.Parent.IsKind(SyntaxKind.ParameterList) &&
+                tokenOnLeftOfPosition.Parent.IsParentKind(SyntaxKind.ParenthesizedLambdaExpression))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public static bool IsPossibleDeconstructionDesignation(this SyntaxTree syntaxTree,
+            int position, CancellationToken cancellationToken)
+        {
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+
+            if (token.Parent.IsKind(SyntaxKind.SingleVariableDesignation))
+            {
+                var designation = (SingleVariableDesignationSyntax)token.Parent;
+                if (designation.IsParentKind(SyntaxKind.DeclarationExpression))
+                {
+                    var declaration = (DeclarationExpressionSyntax)designation.Parent;
+                    return !declaration.Type.IsMissing && designation.Identifier == token;
+                }
+                return false;
             }
 
-            return tokenOnLeftOfPosition.IsKind(SyntaxKind.CommaToken) &&
-                tokenOnLeftOfPosition.Parent.IsKind(SyntaxKind.TupleExpression, SyntaxKind.TupleType);
+            var prev = token.GetPreviousTokenIfTouchingWord(position);
+
+            return IsPossibleTupleContext(syntaxTree, token, position) &&
+                !prev.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken);
```

Comment:
i'm confused by teh " !prev.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken)" part.  Why does the presence of an open paren or comma make this *not* a possible deconsturction designation?